core/src/env_platform/db_handler.bal

```diff
@@ -171,3 +171,106 @@ function updateApplication(TreeRemovalForm form, string applicationId) returns b
+    }
+}
+
+# The `saveApplicationInUser` function will save the application reference in the corresponding user.
+# 
+# + userId - Id of the user.
+# + applicationId - Id of the application saved.
+# + applicationType - Type of the application.
+# + return - This function will return either application is saved in the given user document or 
+# error if there is a mongodb:DatabaseError.
+function saveApplicationInUser(string userId, string applicationId, string applicationType) returns boolean|error {
+
+    boolean isValid = check isValidUser(userId);
+    if (isValid) {
+
+        // Get the user information.
+        map<json>[] find = check usersCollection->find({id: userId});
+        json|error applications = find[0].applications;
+
+        // Construct the applicationList.
+        json[] applicationList;
+        if (applications is error) {
+            applicationList = [{id: applicationId, name: applicationType}];
+        } else {
+            applicationList = <json[]>applications;
+            log:printDebug("The user's applications are: ");
+            applicationList.forEach(function (json value) {
+                log:printDebug(value.toString());
+            });
+            applicationList.push(<json>{id: applicationId, name: applicationType});
+        }
+
+        // Update the user applications array with incoming value.
+        int updated = check usersCollection->update({"applications": applicationList}, {id: userId});
+        log:printDebug("Updated application list is: ");
+        applicationList.forEach(function (json value) {
+            log:printDebug(value.toString());
+        });
+        return updated > 0 ? true : false;
+    } else {
+        return error("Invalid User", message = "Couldn't find the user with given User ID");
+    }
+}
+
+# The `removeApplicationInUser` function will remove the application reference in the corresponding user.
+# 
+# + userId - Id of the user.
+# + applicationId - Id of the application which should be removed.
+# + return - This function will return either application is removed in the given user document or 
+# error if there is a mongodb:DatabaseError.
+function removeApplicationInUser(string userId, string applicationId) returns boolean|error {
+    boolean isValid = check isValidUser(userId);
+    if (isValid) {
+
+        // Get the user information.
+        map<json>[] find = check usersCollection->find({id: userId});
+        json|error applications = find[0].applications;
+
+        if (applications is error) {
+            log:printDebug("The error occured is: " + applications.toString());
+            return error("No applications", message = "There are no applications for the user: " + userId + ".");
+        } else {
+            // Convert json to json array.
+            json[] applicationArray = <json[]>applications;
+            log:printDebug("The user's applications are : ");
+            applicationArray.forEach(function (json value) {
```

Comment:
@yashodgayashan On second thoughts, this will have a significant performance impact in normal use, especially with large application lists. Is there a way to identify when Ballerina is run in debug mode and then run this code only when it is enabled? If yes, then that would be my suggested change. If no it might be better to leave how it previously was for now. We'll figure out a way to improve this in the future.

Reply:
@deshankoswatte @vikumkbv appreciate your thoughts as well

Reply:
@sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications.");` . what do you say this provides more than enough in my opinion.

Reply:
> @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications.");` . what do you think. For me this provides more than enough details.

This is good

Reply:
> @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.

@vikumkbv the details above is enough for us to go to the db and check if there's a problem right?

Reply:
@deshankoswatte Should I change the debug logs according to this? @sanjulamadurapperuma appreciate your suggestions.

Reply:
@deshankoswatte yes that's the best way we can write the debug log. I was just wandering whether we actually need all the application data to be output to the log in debug. @yashodgayashan +1 for the change suggested

Reply:
> > @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.
> 
> @vikumkbv the details above is enough for us to go to the db and check if there's a problem right?

There might be edge cases where it might not work, but for the basic use cases this might do.

Reply:
> @yashodgayashan On second thoughts, this will have a significant performance impact in normal use, especially with large application lists. Is there a way to identify when Ballerina is run in debug mode and then run this code only when it is enabled? If yes, then that would be my suggested change. If no it might be better to leave how it previously was for now. We'll figure out a way to improve this in the future.

@sanjulamadurapperuma you can run ballerina in debug mode using the flag `--b7a.log.level=[LOG_LEVEL]`. In default mode INFO and above will be seen. Check https://ballerina.io/learn/by-example/log-api.html for more info.

Reply:
> > @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.
> 
> @vikumkbv the details above is enough for us to go to the db and check if there's a problem right?

@deshankoswatte i think it is enough, or we can add last db record id to log

Reply:
> > > @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.
> > 
> > 
> > @vikumkbv the details above is enough for us to go to the db and check if there's a problem right?
> 
> There might be edge cases where it might not work, but for the basic use cases this will do.

For that we'll need to scrape the db and see a bit more. I think that's fine.

Reply:
> > @yashodgayashan On second thoughts, this will have a significant performance impact in normal use, especially with large application lists. Is there a way to identify when Ballerina is run in debug mode and then run this code only when it is enabled? If yes, then that would be my suggested change. If no it might be better to leave how it previously was for now. We'll figure out a way to improve this in the future.
> 
> @sanjulamadurapperuma you can run ballerina in debug mode using the flag `--b7a.log.level=[LOG_LEVEL]`. In default mode INFO and above will be seen. Check https://ballerina.io/learn/by-example/log-api.html for more info.

@deshankoswatte yes you can do that, but my concern was whether we can identify that we are running Ballerina in debug mode within the code, so that we can conditionally limit the for loop to run only in debug mode.

Reply:
> @deshankoswatte Should I change the debug logs according to this? @sanjulamadurapperuma appreciate your suggestions.

Move on with my suggestion @yashodgayashan.

Reply:
I will make changes.

Reply:
> > > @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.
> > 
> > 
> > @vikumkbv the details above is enough for us to go to the db and check if there's a problem right?
> 
> @deshankoswatte i think it is enough, or we can add last db record id to log

Last db record... any special reasons for it? It doesn't seem to be necessary to me to only show the last record as a user may have more than one application right? Correct me if I am wrong.

Reply:
> > > @yashodgayashan On second thoughts, this will have a significant performance impact in normal use, especially with large application lists. Is there a way to identify when Ballerina is run in debug mode and then run this code only when it is enabled? If yes, then that would be my suggested change. If no it might be better to leave how it previously was for now. We'll figure out a way to improve this in the future.
> > 
> > 
> > @sanjulamadurapperuma you can run ballerina in debug mode using the flag `--b7a.log.level=[LOG_LEVEL]`. In default mode INFO and above will be seen. Check https://ballerina.io/learn/by-example/log-api.html for more info.
> 
> @deshankoswatte yes you can do that, but my concern was whether we can identify that we are running Ballerina in debug mode within the code, so that we can conditionally limit the for loop to run only in debug mode.

You mean like `if(debugLogsEnabled)` right? I don't think we can do that in Ballerina yet :(

Reply:
> > > > @yashodgayashan On second thoughts, this will have a significant performance impact in normal use, especially with large application lists. Is there a way to identify when Ballerina is run in debug mode and then run this code only when it is enabled? If yes, then that would be my suggested change. If no it might be better to leave how it previously was for now. We'll figure out a way to improve this in the future.
> > > 
> > > 
> > > @sanjulamadurapperuma you can run ballerina in debug mode using the flag `--b7a.log.level=[LOG_LEVEL]`. In default mode INFO and above will be seen. Check https://ballerina.io/learn/by-example/log-api.html for more info.
> > 
> > 
> > @deshankoswatte yes you can do that, but my concern was whether we can identify that we are running Ballerina in debug mode within the code, so that we can conditionally limit the for loop to run only in debug mode.
> 
> You mean like `if(debugLogsEnabled)` right? I don't think we can do that in Ballerina yet :(

Yeah I couldn't find a way to do that in Ballerina either

Reply:
> > > > @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.
> > > 
> > > 
> > > @vikumkbv the details above is enough for us to go to the db and check if there's a problem right?
> > 
> > 
> > @deshankoswatte i think it is enough, or we can add last db record id to log
> 
> Last db record... any special reasons for it? It doesn't seem to be necessary to me to only show the last record as a user may have more than one application right? Correct me if I am wrong.

I think @vikumkbv has meant the log id of the last db operation. I don't think thats also possible with ballerina and mongodb.

Reply:
> > > > @sanjulamadurapperuma we'll change it like `log:printDebug("The user with the user ID: " + userId + " has " + noOfApplications + " applications stored in the database.");` . what do you think. For me this provides more than enough details.
> > > 
> > > 
> > > @vikumkbv the details above is enough for us to go to the db and check if there's a problem right?
> > 
> > @deshankoswatte i think it is enough, or we can add last db record id to log
> 
> Last db record... any special reasons for it? It doesn't seem to be necessary to me to only show the last record as a user may have more than one application right? Correct me if I am wrong.

We'll go with dehami's suggestion